s3-sink-connector/src/main/java/io/aiven/kafka/connect/s3/config/S3SinkConfig.java

```diff
@@ -31,106 +31,36 @@
 import org.apache.kafka.common.config.ConfigDef.Type;
 import org.apache.kafka.common.config.ConfigException;
 
-import io.aiven.kafka.connect.common.config.AivenCommonConfig;
 import io.aiven.kafka.connect.common.config.CompressionType;
-import io.aiven.kafka.connect.common.config.FixedSetRecommender;
+import io.aiven.kafka.connect.common.config.FileNameFragment;
 import io.aiven.kafka.connect.common.config.OutputField;
 import io.aiven.kafka.connect.common.config.OutputFieldEncodingType;
 import io.aiven.kafka.connect.common.config.OutputFieldType;
 import io.aiven.kafka.connect.common.config.TimestampSource;
-import io.aiven.kafka.connect.common.config.validators.FileCompressionTypeValidator;
-import io.aiven.kafka.connect.common.config.validators.FilenameTemplateValidator;
-import io.aiven.kafka.connect.common.config.validators.NonEmptyPassword;
-import io.aiven.kafka.connect.common.config.validators.OutputFieldsValidator;
 import io.aiven.kafka.connect.common.config.validators.TimeZoneValidator;
 import io.aiven.kafka.connect.common.config.validators.TimestampSourceValidator;
-import io.aiven.kafka.connect.common.config.validators.UrlValidator;
 import io.aiven.kafka.connect.common.templating.Template;
+import io.aiven.kafka.connect.config.s3.S3ConfigFragment;
+import io.aiven.kafka.connect.config.s3.S3SinkBaseConfig;
+import io.aiven.kafka.connect.iam.AwsStsEndpointConfig;
+import io.aiven.kafka.connect.iam.AwsStsRole;
 import io.aiven.kafka.connect.s3.S3OutputStream;
 
-import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
 import com.amazonaws.regions.Region;
 import com.amazonaws.regions.RegionUtils;
 import com.amazonaws.regions.Regions;
 import com.amazonaws.services.s3.internal.BucketNameUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-@SuppressWarnings({ "PMD.TooManyMethods", "PMD.GodClass", "PMD.ExcessiveImports" })
-final public class S3SinkConfig extends AivenCommonConfig {
+@SuppressWarnings({ "PMD.TooManyMethods", "PMD.GodClass", "PMD.ExcessiveImports", "PMD.TooManyStaticImports" })
+final public class S3SinkConfig extends S3SinkBaseConfig {
```

Comment:
I cannot see in this pr, but can we delete BucketNameValidator this unused sub class from this

Reply:
The BucketNameValidator now belongs to the S3ConfigFragment, this handles all S3 specific configuration.

Reply:
Yes, but the unused internal subclass also exists in S3SinkConfig.  I think we can delete it

Reply:
Weird! I didn't find that on my search of the class when you pointed it out, but I see it now 😩

Reply:
Also the below 3 methods, hasAwsStsRole(), hasStsEndpointConfig(), getAwsEndpointConfiguration() can be deleted from S3SinkBaseConfig